src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -85,6 +85,17 @@ class PalantirDockerPlugin implements Plugin<Project> {
             dependsOn tag
         })
 
+        Task pushAllTags = project.tasks.create('dockerTagsPush', {
+            group = 'Docker'
+            description = 'Pushes all tagged Docker images to configured Docker Hub.'
+        })
+
+        Task pushAll = project.tasks.create('dockerAllPush', {
```

Comment:
I think these "push all" workflows are already be supported by the `dockerPush` task.

As long as `name` does not include a tag, then `name` is the docker repository and `docker push myname` will push the image and all tags.

Reply:
I think the `dockerPush` just push the main docker, will not push the tags.
`dockerTagsPush` will just push the tags, will not push the main docker.
`dockerAllPush` will push both of the main docker and the tags.

Reply:
This seems reasonable. It is a bit weird since `dockerPush` might behave the same as `dockerAllPush` if the `name` field does not include a tag. But we can't really avoid this with the current design.

This really just highlights how misleading the `name` field is. Ultimately, I think it would be preferable to treat the `name` field like any other `tag` and apply all the tags when building the image (see https://github.com/moby/moby/pull/15780).

A couple concrete comments here:
1. Adding these tasks is unrelated to the improve DSL, can we move this change to a separate PR?
2. What's the use case for `dockerTagsPush`? It seems a bit confusing to me. I think we probably just need something like `dockerPush<tag>` and `dockerPushAll`.

Reply:
First, we can't use `dockerPushAll`, due to it has a conflict with `dockerPush<tag>` when you have a tagging task named 'all'. But maybe we can some names as reserved like 'all', 'tags'?

In fact, I am using `dockerTagsPush` now.
I have two repositories to push, I named the main dock as `imageName:version`.
And I have two tags, `repostiory1/imageName:version` and `repostiory2/imageName:version`, so I named those tag as `repostiory1` and `repostiory2`.
I push them in our CI system by `dockerTagsPush`, and in some case like debug for docker, I push one of them manually by `dockerPushRepostiory1`.

I will never push the main docker which defined by the name; it could be some confusion if I set the repository to name and using `dockerPush` to push one.
Why I push repository1 using `dockerPushRepostiory1`, but repository2 is `dockerPush`? Should it be `dockerPushRepostiory2`?

So I create the `dockerTagsPush` and `dockerAllPush` task.

Reply:
I would be in favor of adding `dockerTagsPush` but not `dockerAllPush`. This will allow us to move closer toward making `name` and `tags` behave the same (since under the hood they are both just tags).

So the image tags can be set and pushed by either:
- Using the `name` field and the `dockerPush` task
- Using the `tag` field and either `dockerPushMyTaskName` or `dockerTagsPush`

Reply:
OK, got it